scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java

```diff
@@ -0,0 +1,301 @@
+                                                                           "groovy",
+                                                                           TargetType.NODE_URL.name(),
+                                                                           Sets.newSet(testNode.getNodeURL()));
+        Assert.assertEquals(1, results.size());
+        ScriptResult result = results.get(0);
+        if (result.errorOccured()) {
+            result.getException().printStackTrace();
+            Assert.fail("Script execution failed");
+        }
+
+        log("Let task finish");
+        fileLock.unlock();
+
+        log("Wait when job finish");
+        schedulerHelper.waitForEventJobFinished(jobId, TIMEOUT);
+
+        schedulerHelper.waitUntilState(testNode.getNodeURL(), NodeState.FREE, TIMEOUT);
+
+        log("Check job result");
+        checkJobResult(schedulerHelper.getSchedulerInterface(), jobId);
+
+        Assert.assertEquals("After the node is freed, no active object should remain",
+                            0,
+                            testNode.getNode().getNumberOfActiveObjects());
+    }
+
+    @Test
+    public void testRunningTaskWithSelectionScriptDoesNotImpactScriptExecution() throws Exception {
+        testRunningTaskDoesNotImpactScriptExecution(true);
+    }
+
+    @Test
+    public void testRunningTaskWithoutSelectionScriptDoesNotImpactScriptExecution() throws Exception {
+        testRunningTaskDoesNotImpactScriptExecution(false);
+    }
+
+    public void testRunningTaskDoesNotImpactScriptExecution(boolean withSelectionScript) throws Exception {
+        FileLock fileLock = new FileLock();
+        Path fileLockPath = fileLock.lock();
+
+        String scriptCode = "org.ow2.proactive.scheduler.util.FileLock.waitUntilUnlocked(\"" +
+                            fileLockPath.toString().replace("\\", "\\\\") + "\")";
+
+        log("Submit script (handled asynchronously by the resource manager)");
+        List<ScriptResult<Object>> resultsFuture = schedulerHelper.getResourceManager()
+                                                                  .executeScript(scriptCode,
+                                                                                 "groovy",
+                                                                                 TargetType.NODE_URL.name(),
+                                                                                 Sets.newSet(testNode.getNodeURL()));
+
+        log("Submit job");
+        final JobId jobId = schedulerHelper.submitJob(createNonBlockingJob(withSelectionScript));
+
+        // We wait a bit to be sure the job went through the scheduling loop
+        // if a selection script is set, the task start will be delayed as much as the SelectionManager is busy executing this script
+        // (and thus cannot execute the selection script provided by the task)
+        Thread.sleep(3000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)